src/ngraph/validation_util.hpp

```diff
@@ -185,6 +185,14 @@ namespace ngraph
     /// \return (true, max_value) if can be determined, or (false, 0) if not.
     NGRAPH_API std::pair<bool, int64_t> maximum_value(const Output<Node>& value);
 
+    /// \brief Evaluates values, treating values in value_map as already computed. value_map is
+    /// updated.
+    /// \param value_map Key is RawNodeOutput in graph, value is the computed value. Updated by the
+    /// function.
+    /// \param values Root set of values to try to compute
+    NGRAPH_API void evaluate_nodes(std::map<RawNodeOutput, Output<Node>>& value_map,
```

Comment:
ngraph::Node has `constant_fold` (or better `evaluate`) method. Why not to attach the evaluation capability to ngraph::Lambda or ngraph::Function directly as a class method?

Reply:
If you do that, then you can only use it on the entire graph. This way, there could be a method on Lambda that calls the generic evaluator, but it's still available for use in other transformations.
Note that custom ops can participate in evaluation.

Reply:
We discussed it during our meeting. If it is easy to do, please provide a method. If it requires more time, please postpone to some other PR. I want to finalize this PR with core stuff ASAP.

Reply:
Lambda is in now, but it won't work until all the ops in the lambda have evaluate methods.